mlrun/feature_store/steps.py

```diff
@@ -226,34 +227,96 @@ def _do_pandas(self, event):
+                except AnalysisException:
+                    df = df.withColumn(
+                        new_column_name, mapping_expr.getItem(col(column))
+                    )
+                    col_type = df.schema[column].dataType
+                    new_col_type = df.schema[new_column_name].dataType
+                    #  in order to avoid exception at isna on non-decimal/float columns -
+                    #  we need to check their types before filtering.
+                    if isinstance(col_type, (FloatType, DoubleType, DecimalType)):
+                        column_filter = (~isnull(col(column))) & (~isnan(col(column)))
+                    else:
+                        column_filter = ~isnull(col(column))
+                    if isinstance(new_col_type, (FloatType, DoubleType, DecimalType)):
+                        new_column_filter = isnull(col(new_column_name)) | isnan(
+                            col(new_column_name)
+                        )
+                    else:
+                        #  we need to check that every value replaced if we changed column type - except None or NaN.
+                        new_column_filter = isnull(col(new_column_name))
+                    mapping_to_null = [
+                        k
+                        for k, v in column_map.items()
+                        if v is None
+                        or (
+                            isinstance(v, (float, np.float64, np.float32, np.float16))
+                            and math.isnan(v)
+                        )
+                    ]
+                    turned_to_none_values = df.filter(
+                        (column_filter) & (new_column_filter)
+                    ).filter(~col(column).isin(mapping_to_null))
+
+                    if not turned_to_none_values.isEmpty():
+                        raise mlrun.errors.MLRunInvalidArgumentError(
+                            f"Mapvalues - mapping that changing column type must change all values accordingly"
+                            f" the column! Column- {column}"
+                        )
             else:
                 for val, val_range in column_map["ranges"].items():
                     min_val = val_range[0] if val_range[0] != "-inf" else -np.inf
                     max_val = val_range[1] if val_range[1] != "inf" else np.inf
                     otherwise = ""
-                    if new_column_name in event.columns:
-                        otherwise = event[new_column_name]
-                    event = event.withColumn(
+                    if new_column_name in df.columns:
+                        otherwise = df[new_column_name]
+                    df = df.withColumn(
                         new_column_name,
                         when(
-                            (event[column] < max_val) & (event[column] >= min_val),
+                            (df[column] < max_val) & (df[column] >= min_val),
```

Comment:
According to PyCharm at least, the `&` here is resolving to the python built in implementation,  not the pyspark one. If this is true, it's unlikely that this line will work (is there test coverage for this case?)

Reply:
I still see the same warning. How was this resolved?

Reply:
Ok, must be pycharm that's in the wrong then.